oeps/oep-0019-bp-developer-documentation.rst

```diff
@@ -0,0 +1,450 @@
+
+* **Version Controlled**. Documentation must be maintained under version control in Github, for the
+  following reasons:
+
+  * **Reviewed**. They will be properly reviewed as defined by that repo's process. Anyone in the Open edX
+    community with access to the repo will have the ability to add, update, fix, and contribute to
+    documentation by following the repo's review and maintainence process.
+
+  * **Tagged**. They will be appropriately in-synch with their corresponding code. For example,
+    when an open edX instance runs an older named release, the operator can refer to the matching
+    documentation for that version of the codebase.
+
+  * **Source Controlled**. They will have the advantages of having source-control backed files, including sharing
+    across multiple developers and robust storage.
+
+.. _edx-developer-docs: https://github.com/edx/edx-developer-docs
+.. _docs.edx.org: https://docs.edx.org/
+
+Format
+======
+
+We will use `reStructuredText (rST)`_ for all developer documentation.
+
+* rST scales up: it can be used for anything from docstrings to single-page
+  documents to books to multi-volume book sets.
+
+* rST has a rich feature set, including inter-document links, table of
+  contents, auto-collection of code documentation, list-formatted tables,
+  semantic markup, and multiple output formats.
+
+* A single documentation format enables consistency.  It also avoids conversion
+  to rST when we require its additional features.  Our experience has shown
+  that these two benefits of standardization keep down the overall cost of
+  documentation maintenance.
+
+* rST is the standard markup format for Python project documentation.
+
+* rST does not suffer from incompatible implementations, with different
+  features supported differently by different renderers.
+
+* GitHub renders rST natively.
+
+More details on these points are in Eric Holscher's `why rST over Markdown for
+documentation`_ blog post.
+
+.. _why rST over Markdown for documentation: http://ericholscher.com/blog/2016/mar/15/dont-use-markdown-for-technical-docs/
+
+
+Best practices
+--------------
+
+* We strongly recommend keeping line length between 80 and 120 characters. Keeping it less than 120
+  leaves room for future edits.
+
+* Consistency across rST documents is not a requirement.  However, you can use the raw version of this OEP
+  as a reference example.
+
+Learning rST
+------------
+
+* If you are familiar with Markdown:
+  
+  * Here is edX's `cheatsheet comparing rST to Markdown`_.
+
+  * Here is `another comparison of Markdown and rST`_.
+
+* See the `reStructuredText (rST) Primer`_.
+
+.. _cheatsheet comparing rST to Markdown: https://github.com/edx/mdrst/blob/master/mdrst.rst
+.. _another comparison of Markdown and rST: https://gist.github.com/dupuy/1855764
+
+
+Helpful rST-related tools
+-------------------------
+
+* `Visual Studio Code`_ works pretty well.
+
+* PyCharm_ provides an rST editor and preview panel. 
+
```

Comment:
There's a rST preview package for Atom as well: https://atom.io/packages/rst-preview-pandoc

![image](https://user-images.githubusercontent.com/17170027/51094395-8406d500-177a-11e9-9089-94e92f50e44f.png)